pkg/clean/types.go

```diff
@@ -3,3 +3,27 @@ package clean
 type cleanup interface {
 	Cleanup() error
 }
+
+type CleanupService struct {
+	cleans []cleanup
+	err    error
+}
+
+func NewCleanupService() *CleanupService {
+	return &CleanupService{}
+}
+
+func (c *CleanupService) AddCleanupService(f func() (cleanup, error)) *CleanupService {
+	var clean cleanup
+	if c.err != nil {
+		return c
+	}
+
+	clean, c.err = f()
+	if c.err != nil {
+		return c
+	}
```

Comment:
does that mean if one service fail to init, the cleanup could still continue without report a error? or we should break the cleanup procedure and quit? or retry some time here to quit?

Reply:
ditto